orca-core/src/main/java/com/netflix/spinnaker/orca/locks/LockContext.java

```diff
@@ -54,12 +55,21 @@ public LockValueBuilder withStage(Stage stage) {
             .map(Execution.ExecutionType::toString)
             .orElse(null));
 
-        //TODO(cfieber): lockValue.id:
-        // if we are a child pipeline / strategy we probably need to use the parent execution id
-        final String id = Optional.ofNullable(this.id).orElseGet(() ->
-          execution
-            .map(Execution::getId)
-            .orElse(null));
+        final String id = Optional.ofNullable(this.id).orElseGet(() -> {
+          if (!execution.isPresent()) {
+            return null;
+          }
+
+          final Execution exec = execution.get();
+          if (exec.getTrigger() instanceof PipelineTrigger) {
+            return Optional.ofNullable(((PipelineTrigger) exec.getTrigger()).getParentStage())
```

Comment:
Can a strategy have an embedded strategy?

Reply:
it can't - the deploy stage in the strategy is just a placeholder for where to inject the cluster context from the parent

the child pipeline however, that probably totally maybe can...